src/Symfony/Component/Cache/Adapter/ApcuAdapter.php

```diff
@@ -27,22 +30,49 @@ public function __construct($namespace = '', $defaultLifetime = 0)
             ini_set('apc.use_request_time', 0);
         }
         parent::__construct($namespace, $defaultLifetime);
+        $this->nonce = (string) $nonce;
+        $this->nonceLen = strlen($nonce);
     }
 
     /**
      * {@inheritdoc}
      */
     protected function doFetch(array $ids)
     {
-        return apcu_fetch($ids);
+        $values = apcu_fetch($ids);
+
+        if ($this->nonceLen) {
+            foreach ($values as $id => $v) {
+                if (!is_string($v) || 0 !== strpos($v, $this->nonce)) {
+                    apcu_delete($id);
+                    unset($values[$id]);
+                }
+            }
+            foreach ($values as $id => $v) {
+                $values[$id] = unserialize(substr($v, $this->nonceLen));
```

Comment:
Good question: because unserialization involves userland code, which means can be slow. But IMHO we need to invalidate fast.